NewHaven/src/Square.cpp

```diff
@@ -18,9 +18,7 @@ Square::Square() : visitedResource{ new bool[4]{ false, false, false, false } },
 Square::Square(const Square& square) : visitedResource{ new bool[4] }, rootConnected{ new bool[4] }
 {
     if(square.tile != nullptr) {
-        tile = new HarvestTile();
-        // use the operator overloader
-        *tile = *square.tile;
+        tile = new HarvestTile(*square.tile);
     }
     else
         tile = nullptr;
```

Comment:
No copying harvest tiles, assign it straight, you won't need to make the nullptr check that way too. I didn't check, but square should not delete the tile in its destructor, if you copy it then you'll have to, but we won't. It's ok to have a square and its copy point to the same harvest tile as long as they don't delete it.

Reply:
That means you will need to modify your resource score algorithm to reset your other fields since they will no longer be copies.